### ISSUE TITLE
Image "field" templates showing up when creating new page

### ISSUE THREAD

**Issue by adrianbj** [processwire/processwire-issues#1524]
### Short description of the issue

If you create a new page under a parent/template that allows multiple templates (no child template restrictions), templates like 'field-image`, `field-images` etc appear in the selectable list.

### Expected behavior

These "field" templates shouldn't be shown.

### Optional: Screenshots/Links that demonstrate the issue

![image](https://user-images.githubusercontent.com/871211/153313395-d4ec5e2e-0329-481c-9562-80169d4686a0.png)

**Comment by BernhardBaumrock** [comment]
shouldnt those templates just be flagged as system templates? or are they and nonetheless show up there? i've never used them so I can't check, but I've built a module where I'm hiding templates by setting the system flag...

**Comment by adrianbj** [comment]
> shouldnt those templates just be flagged as system templates

Yes, I would think that they should be considered system templates.

**Comment by elabx** [comment]
Hi! Just want to bring attention to this issue again, I feel these templates should be treated exactly like `repeater_` templates, make them system templates on creation if there is a field that matches the name pattern.

If family settings for templates are not taken care care properly on multiple template sites it becomes messy when adding new pages or when changing the template of a page, since I normally leave these decisions to the user, the `field-{fieldname}` named templates break this very organized editing experience. It also doesn't really make sense to configure them in family settings, right?

Thanks for reading!

**Comment by matjazpotocnik** [comment]
Setting Template::flagSystem alone does not hide a template from the dropdowns. It does hide it in the templates list at /processwire/setup/temples/ and that's wrong. In ProcessPageAdd templates are excluded if they have `noParents=1`. In ProcessPageEdit it checks flagSystem, but ONLY explicitly blocks names starting with` repeater_` and `fieldset_`. Possible workaround:

```php
$wire->addHookBefore('Templates::saveReady', function(HookEvent $event) {
    $template = $event->arguments(0);
    
    if ($template instanceof Template && strpos($template->name, 'field-') === 0) {
        $template->noParents = 1;
        //$template->noChildren = 1;
        //$template->noGlobal = 1;
    }
});
```

For existing templates you would have to save them.

@ryancramerdesign would you consider fixing this?

**Comment by ryancramerdesign** [comment]
Thanks Adrian for the report, and thanks Matjaz for identifying `noParents` as the appropriate template setting. This is now fixed on the dev branch in processwire/processwire@13cf987a. Newly created custom file/image field templates receive `noParents=1`, preventing them from appearing as templates for page creation or template changes. The FieldtypeFile module upgrade also updates existing `field-*` and `field-x-*` templates associated with file or image fields, while requiring the core-created `noGlobal` marker so similarly named user templates are left alone. Regression coverage includes both file and image fields, and the focused FieldtypeFile and FieldtypeImage tests pass.
